storm-core/src/clj/org/apache/storm/daemon/nimbus.clj

```diff
@@ -1793,16 +1793,16 @@
         (.mark nimbus:num-downloadChunk-calls)
         (check-authorization! nimbus nil nil "fileDownload")
         (let [downloaders (:downloaders nimbus)
-              ^BufferFileInputStream is (.get downloaders id)]
+              ^BufferInputStream is (.get downloaders id)]
```

Comment:
@HeartSaVioR, in beginFileDownload (just above the impl for downloadChunk) the type used for the input stream and put into :downloaders is a BufferInputStream, hence the change in downloadChunk.